### PR TITLE
Useridtracking

### LINKED ISSUE #28: Track userID to username mapping
Telegram users can change their usernames, but not their userIDs. We use userIDs to keep track of who is who. However, when sending text for humans to read, we need to transform it back to usernames.

1. Create table to store this data
2. Create method to update data every time a user interacts with the bot 
3. Create method to get username from userID

### PR DESCRIPTION
**Pull request by gyulong1**
Updated user_id tracking without a handler file

Error handling now in queries.js

### CODE REVIEW

**Review by nicktohzyu — APPROVED** [review]
Approved, however I did not test with postgres. Usage of these helpers will require updating the database

File: db/queries.js [hunk 1]
```diff
@@ -198,6 +198,41 @@ module.exports.getItemName = async function (params, callback) {
...
+        const statement = `
+		insert into 
+		miscellaneous.usernames (user_name, user_id)
+		values ($1, $2)
+        on conflict (user_id) do update
+		set user_name = $3;`;
+        let args;
+        //checks if username is empty
+        if (params.user_name === undefined) {
+           args = ["", params.user_id, ""];
```
Comment (nicktohzyu) [inline db/queries.js]: 3 spaces for indentation ‽

File: db/queries.js [hunk 2]
```diff
@@ -198,6 +198,41 @@ module.exports.getItemName = async function (params, callback) {
...
+        } else{
```
Comment (nicktohzyu) [inline db/queries.js]: Please fix brace spacing - use the automatic formatting tool

File: db/queries.js [hunk 3]
```diff
@@ -198,6 +198,41 @@ module.exports.getItemName = async function (params, callback) {
...
+}
+
+module.exports.updateUsername = async function (params, callback) {
+    try{
+        const statement = `
+		insert into 
+		miscellaneous.usernames (user_name, user_id)
+		values ($1, $2)
+        on conflict (user_id) do update
+		set user_name = $3;`;
```
Comment (nicktohzyu) [inline db/queries.js]: Would $1 work instead of $3?

File: db/queries.js (hunk 1 above)
Comment (nicktohzyu) [reply db/queries.js]: I'd suggest to `params.user_name = ""` instead. This makes it act like a guard clause. Then the line inside the else block will be moved outside (and the else block will be empty, hence remove it)

**Review by nicktohzyu — CHANGES_REQUESTED** [review]

**Comment by nicktohzyu** [comment]
Please update on receiving a callback as well. In `index.js`, line 92

File: db/queries.js (hunk 3 above)
Comment (gyulong1) [reply db/queries.js]: yes it does work, thanks

File: db/queries.js (hunk 1 above)
Comment (gyulong1) [reply db/queries.js]: done, thanks

File: db/queries.js (hunk 2 above)
Comment (gyulong1) [reply db/queries.js]: done

**Review by nicktohzyu — APPROVED** [review]

**Pull request merged**